models.py

```diff
@@ -1401,3 +1403,36 @@ def __init__(self, *args, **kwargs):
 			nonlinearity = ('leaky_relu', 0.01),
 			**kwargs
 		)
+
+
+class OnnxModelWraper(nn.Module):
+	def __init__(self, onnx_model_path, provider=None):
+		self.dict = dict
+		self.bpe_only = False
+		super(OnnxModelWraper, self).__init__()
+		onnxruntime_session = onnxruntime.InferenceSession(onnx_model_path)
+
+		if provider:
+			onnxruntime_session.set_providers(list(provider))
+
+		self.model = lambda x: onnxruntime_session.run(None, dict(x=x))
+
+	def forward(self, x, xlen, y, ylen):
+		load_batch = lambda x: x.cpu().numpy()
+		logits = self.model(load_batch(x.squeeze(1)))
+
+		print('argmax: ', torch.from_numpy((logits[0])).argmax(dim=1))
+		logits = [torch.from_numpy(l).to(device=y.device) for l in logits]
```

Comment:
please fuse `torch.from_numpy(...)` and `to(...)` into a single `torch.as_tensor(...)` call

Reply:
done